### DISCUSSION TITLE
500 Internal Server Error While Attempting to Create Private Repository

### DISCUSSION THREAD

**Discussion by kdelorey** [community/community#40491] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

While playing around with the GitHub Terraform provider, I discovered that attempts to create a private GitHub repository results in a 500 Internal Server Error. Note this isn't specific Terraform, just how I discovered the issue.

If try to make the same request Terraform attempts to do in any HTTP client (eg. `curl`) the same result occurs. For example, a `POST` to `https://api.github.com/user/repos` with the following body results in the error:

```json
{
 "name": "terraform-test",
 "description": "A test.",
 "homepage": "",
 "private": true,
 "visibility": "private",
 "has_issues": false,
 "has_projects": false,
 "has_wiki": false,
 "is_template": false,
 "auto_init": false,
 "gitignore_template": "",
 "license_template": "",
 "allow_squash_merge": true,
 "allow_merge_commit": true,
 "allow_rebase_merge": true,
 "allow_auto_merge": false,
 "delete_branch_on_merge": false,
 "squash_merge_commit_title": "COMMIT_OR_PR_TITLE",
 "squash_merge_commit_message": "COMMIT_MESSAGES",
 "merge_commit_title": "MERGE_MESSAGE",
 "merge_commit_message": "PR_TITLE"
}
```

After experimenting with it for a little while, it seems you can change the `"private": true` boolean value into a string like `"private": "true"`, and the API successfully create a private repo. However [according to the API documentation](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#create-a-repository-for-the-authenticated-user), `private` is expected to be a `boolean` type, with a provided example of it being a boolean too. So I don't think its appropriate to file an issue with the TF provider project, as it is making the request correctly as I understand it.

I should also note that changes to the `visibility` property, including removing it completely, has no effect when making the request. The API doesn't actually expect this property to be present when making an user repository, but the TF provider puts it in the request regardless.

**Comment by ossanna16** [comment]
Hi there @kdelorey and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by mvarchdev** [comment]
@kdelorey do you have solution for that?

**Reply by kdelorey** [reply]
I never found the solution, sorry.
